sessionctx/variable/session.go

```diff
@@ -2318,6 +2322,10 @@ const (
 	SlowLogIsWriteCacheTable = "IsWriteCacheTable"
 )
 
+// GenerateBinaryPlan decides whether we should record binary plan in slow log and stmt summary.
+// It's controlled by the global variable `tidb_generate_binary_plan`.
+var GenerateBinaryPlan atomic2.Bool
```

Comment:
it's a instance scope or cluster scope?

Reply:
It's cluster scope.
It's controlled by the global scope system variable. Once the system variable is updated, the `GenerateBinaryPlan` will be updated in all tidb.